### PR TITLE
FIX: crash: playerfactory is not initialized when using profile login

### PR DESCRIPTION
**Pull request by koying**
Execute the full stage3 when coming from login screen as well

### CODE REVIEW

**Comment by garbear** [comment]
Hehe, another dev joins the login window debacle :)

Throwing more frogs into the pot might fix the problem today, but issues like this will keep popping up until we address the root problem - the broken interdependencies in our codebase.

The proper solution needs to untangle these dependencies so they no longer rely on the login window. `CGUIWindowLoginScreen::LoadProfile()` needs to go.

To accomplish this, we need to first encapsulate all the globals this function uses into service manager. I.E. #13443 and [08dfd8281](https://github.com/garbear/xbmc/commit/08dfd8281). Then we move all these components to the profiles code ([6d236d4b5](https://github.com/garbear/xbmc/commit/6d236d4b5)). Then we invert the dependencies, so instead of profiles depending on all these components, the components register themselves with profiles. That way, we can load and unload profiles and make sure that all the callbacks are called. After some more refactoring, we'll finally have all our services depend on profiles instead of the other way around, such that they can react properly to profile loading/unloading.

I'm at Embedded Linux Conference this week, so feel free to continue where I've left off. Otherwise I'll resume working on this next week.

**Comment by koying** [comment]
Fair enough. I pinged you a couple time regarding but you never ponged back. TZ issues most probably.

May I still suggest that you consider this quick fix to unblock profile users and alleviate some pressure, while the big refactoring (which will no doubt bring its own set of issues) is being put in place?

My long experience as a professional has taught me "better fix today than wait for a better tomorrow" ;)

**Comment by garbear** [comment]
You make a good point. Merge and let the fireworks fly!

File: xbmc/windows/GUIWindowLoginScreen.cpp [hunk 1]
```diff
@@ -338,11 +338,10 @@ void CGUIWindowLoginScreen::LoadProfile(unsigned int profile)
 
   JSONRPC::CJSONRPC::Initialize();
 
-  // Restart context menu manager
-  CServiceBroker::GetContextMenuManager().Init();
-
-  // restart PVR services
-  CServiceBroker::GetPVRManager().Init();
+  if (!g_application.m_ServiceManager->InitStageThree())
```
Comment (ksooo) [inline xbmc/windows/GUIWindowLoginScreen.cpp]: As far as I can see IniStageThree calls `CPeripherals::Initialise()`, but there seems to be no call to `CPeripherals::Clear()` (which is the counterpart to Initialise() - see DeinitStageThree). No good, if you ask me...

Comment (ksooo) [inline xbmc/windows/GUIWindowLoginScreen.cpp]: As far as I can see IniStageThree calls CPeripherals::Initialise(), but there seems to be no call to CPeripherals::Clear() (which is the counterpart of Initialise() - see DeinitStageThree) before this. No good, if you ask me...

**Comment by ksooo** [comment]
Now crashes when changing profile due to CPeripherals inited without being deinited before:

screen shot 2018-03-12 at 23 54 21

**Comment by garbear** [comment]
@koying thanks for this PR, even if it set off fireworks (like I predicted a hack could do). It must have taken time to diagnose and create this PR, which was time you didn't have to give us. I havent been around the last few days, but I image you probably got some blame at some point. I just want to make sure it's balanced out by some recognition that your time is appreciated.

**Pull request merged**